core/src/main/scala/vapors/data/FactTypeSet.scala

```diff
@@ -6,6 +6,9 @@ import cats.data.{NonEmptyList, NonEmptyMap, NonEmptySet}
 
 import scala.collection.immutable.{SortedMap, SortedSet}
 
+// TODO: Is there a better way to do this?
+//       Maybe we can use something like the Has[_] data type to combine multiple fact types?
+
```

Comment:
As far as I can tell, doing a heterogeneous set of fact types would be a pretty big feat. I don't think I can tackle this before launching 1.0.0-M1